include/stdx/optional.hpp

```diff
@@ -186,7 +194,9 @@ template <typename T, typename TS = tombstone_traits<T>> class optional {
 
     [[nodiscard]] friend constexpr auto operator<(optional const &lhs,
                                                   optional const &rhs) -> bool {
-        return lhs.val < rhs.val and rhs.has_value();
+        return lhs.has_value() and rhs.has_value()
+                   ? lhs.val < rhs.val
+                   : not lhs.has_value() and rhs.has_value();
```

Comment:
question: an optional without a value is always considered less than an optional with any value?

Reply:
that's right.